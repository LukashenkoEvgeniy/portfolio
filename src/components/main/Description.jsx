import React from 'react';
import styled, {keyframes} from 'styled-components';

export default function Description ({planet}) {
        return (
            <DesciptionWrapper>
                <Header>{planet.name}</Header>
                <Text>
                    {planet.description}
                </Text>
            </DesciptionWrapper>
        );
}

const Text = styled.div`
  border: darkcyan solid 1px;
  padding: 5px;
  background: linear-gradient(to top left, rgba(0,139,139,0.5), rgba(53,70,39,0.5));;


`;
const Header = styled.div`
    border: darkcyan solid 1px;
    border-radius: 0 70% 0 70%;
    background: linear-gradient(to top left, rgba(0,139,139,0.5), rgba(53,70,39,0.5));;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 50%;
    height: 40px;
`;

const show = keyframes`
  from {
    left: -400px;
  }

  to {
        left: 100px;
  }
`;

const DesciptionWrapper = styled.div`
  position: fixed;
  top: 50px;
  left: 100px;
  display: flex;
  color: white;
  height: 100% - 200px;
  width: 300px;
  overflow-x: auto;
  flex-direction: column;
  align-items: center;
  animation: ${show} 5s linear;
`;
