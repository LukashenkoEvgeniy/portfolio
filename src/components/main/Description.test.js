import Description from "./Description";
import React from "react";

describe('Description test', () => {

    const planet = {
        id: 0,
        name: 'SUN',
        img: null,
        description: 'Сонце — типова зоря головної послідовності спектрального класу G2. Воно майже ідеально сферичне і являє собою гарячу плазму, сплетену магнітними полями. Сонячне випромінювання підтримує життя на Землі (фотони необхідні для початкових стадій процесу фотосинтезу) та визначає клімат нашої планети. Сонце — потужне джерело енергії, яку воно постійно випромінює в усіх ділянках спектра електромагнітних хвиль — від рентгенівських і ультрафіолетових променів до радіохвиль. Це випромінювання сильно впливає на всі тіла Сонячної системи: нагріває їх, позначається на атмосферах планет, дає світло й тепло.'
    };

    it('should render a label', () => {
        const wrapper = shallow(
            <Description planet={planet}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a planet', () => {
        const wrapper = mount(
            <Description planet={planet}/>
        );
        expect(wrapper.prop('planet').id).toEqual(0);
    });
});