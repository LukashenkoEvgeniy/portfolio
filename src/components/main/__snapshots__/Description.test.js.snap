// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description test should render a label 1`] = `
<styled.div>
  <styled.div>
    SUN
  </styled.div>
  <styled.div>
    Сонце — типова зоря головної послідовності спектрального класу G2. Воно майже ідеально сферичне і являє собою гарячу плазму, сплетену магнітними полями. Сонячне випромінювання підтримує життя на Землі (фотони необхідні для початкових стадій процесу фотосинтезу) та визначає клімат нашої планети. Сонце — потужне джерело енергії, яку воно постійно випромінює в усіх ділянках спектра електромагнітних хвиль — від рентгенівських і ультрафіолетових променів до радіохвиль. Це випромінювання сильно впливає на всі тіла Сонячної системи: нагріває їх, позначається на атмосферах планет, дає світло й тепло.
  </styled.div>
</styled.div>
`;

exports[`should render a label 1`] = `
<styled.div>
  <styled.div>
    SUN
  </styled.div>
  <styled.div>
    Сонце — типова зоря головної послідовності спектрального класу G2. Воно майже ідеально сферичне і являє собою гарячу плазму, сплетену магнітними полями. Сонячне випромінювання підтримує життя на Землі (фотони необхідні для початкових стадій процесу фотосинтезу) та визначає клімат нашої планети. Сонце — потужне джерело енергії, яку воно постійно випромінює в усіх ділянках спектра електромагнітних хвиль — від рентгенівських і ультрафіолетових променів до радіохвиль. Це випромінювання сильно впливає на всі тіла Сонячної системи: нагріває їх, позначається на атмосферах планет, дає світло й тепло.
  </styled.div>
</styled.div>
`;
